src/GafferUI/OpenColorIOAlgo.cpp

```diff
@@ -0,0 +1,359 @@
+	{
+		throw IECore::Exception("OpenColorIOAlgo : Initializing LUT : Missing texture data");
+	}
+
+	GLint internalformat = GL_RGB32F_ARB;
+	GLenum format = GL_RGB;
+
+	if (channel == OCIO_NAMESPACE::GpuShaderCreator::TEXTURE_RED_CHANNEL)
+	{
+		internalformat = GL_R32F;
+		format = GL_RED;
+	}
+
+	glGenTextures(1, &texId);
+
+	glActiveTexture(GL_TEXTURE0 + index);
+
+	if (height > 1)
+	{
+		glBindTexture(GL_TEXTURE_2D, texId);
+
+		SetTextureParameters(GL_TEXTURE_2D, interpolation);
+
+		glTexImage2D(GL_TEXTURE_2D, 0, internalformat, width, height, 0, format, GL_FLOAT, values);
+	}
+	else
+	{
+		glBindTexture(GL_TEXTURE_1D, texId);
+
+		SetTextureParameters(GL_TEXTURE_1D, interpolation);
+
+		glTexImage1D(GL_TEXTURE_1D, 0, internalformat, width, 0, format, GL_FLOAT, values);
+	}
+}
+
+} // namespace
+
+namespace GafferUI
+{
+
+namespace OpenColorIOAlgo
+{
+
+IECoreGL::Shader::SetupPtr displayTransformToFramebufferShader( const OCIO_NAMESPACE::Processor *processor )
+{
+	std::string colorTransformCode;
+	OCIO_NAMESPACE::GpuShaderDescRcPtr shaderDesc;
+	if( processor )
+	{
+		shaderDesc = OCIO_NAMESPACE::GpuShaderDesc::CreateShaderDesc();
+		shaderDesc->setLanguage( OCIO_NAMESPACE::GPU_LANGUAGE_GLSL_1_3 );
+		shaderDesc->setFunctionName( "OCIODisplay" );
+
+		OCIO_NAMESPACE::ConstGPUProcessorRcPtr gpuProc = processor->getOptimizedGPUProcessor( OCIO_NAMESPACE::OPTIMIZATION_VERY_GOOD );
+		gpuProc->extractGpuShaderInfo(shaderDesc);
+
+		colorTransformCode = shaderDesc->getShaderText();
+	}
+	else
+	{
+		colorTransformCode = "vec4 OCIODisplay(vec4 inPixel) { return inPixel; }\n";
+	}
+
+	colorTransformCode +=
+	"vec4 luminance( vec4 c )"
+	"{"
+	"	return vec4( vec3( c.r * 0.2126 + c.g * 0.7152 + c.b * 0.0722 ), c.a );\n"
+	"}\n"
+	
+	"vec4 colorTransformWithSolo( vec4 inPixel, int soloChannel )"
+	"{"
+	"	if( soloChannel == -1 )"
+	"	{"
+	"		return OCIODisplay( inPixel );\n"
+	"	}"
+	"	else if( soloChannel == 0 )"
+	"	{"
+	"		return OCIODisplay( inPixel.rrrr ).rrrr;\n"
+	"	}"
+	"	else if( soloChannel == 1 )"
+	"	{"
+	"		return OCIODisplay( inPixel.gggg ).gggg;\n"
+	"	}"
+	"	else if( soloChannel == 2 )"
+	"	{"
+	"		return OCIODisplay( inPixel.bbbb ).bbbb;\n"
+	"	}"
+	"	else if( soloChannel == 3 )"
+	"	{"
+	"		return inPixel.aaaa;\n"
+	"	}"
+	"	else" // -2 is for luminance
+	"	{"
+	"		return luminance( OCIODisplay( luminance( inPixel ) ) );\n"
+	"	}"
+	"}\n"
+
+	"vec4 colorTransform( vec4 inPixel, bool unpremultiply, int soloChannel )"
+	"{"
+	"	if( inPixel == vec4( 0 ) )"
+	"	{"
+	"		return inPixel;\n"
+	"	}"
+	"	else if( inPixel.a == 0 || !unpremultiply || soloChannel == 3 )"
+	"	{"
+	"		return colorTransformWithSolo( inPixel, soloChannel );\n"
+	"	}"
+	"	else"
+	"	{"
+	"		return inPixel.a * colorTransformWithSolo( inPixel / inPixel.a, soloChannel );\n"
+	"	}"
+	"}\n";
+
+	// Build and compile GLSL shader
+	static const char *g_vertexSource =
+	"void main()"
+	"{"
+	"   gl_Position = gl_Vertex;"
+	"   gl_TexCoord[0] = gl_MultiTexCoord0;"
```

Comment:
Can we avoid the legacy pipeline here please? GafferSceneUI/OutputBuffer.cpp shows that it's pretty trivial for this simple use case.